src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/ComplexityOfNeedService.kt

```diff
@@ -2,21 +2,40 @@ package uk.gov.justice.digital.hmpps.keyworker.services
 
 import com.microsoft.applicationinsights.TelemetryClient
 import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.keyworker.events.ComplexityOfNeedLevel
 
 @Service
 class ComplexityOfNeedService(
   val keyworkerService: KeyworkerService,
+  val complexityOfNeedAPI: ComplexityOfNeedAPI,
+  @Value("\${womens.estate}") val enabledPrisons: Set<String>,
   val telemetryClient: TelemetryClient
 ) {
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
+  fun getComplexOffenders(prisonId: String, offenders: Set<String>): Set<String> {
+    if (!enabledPrisons.contains(prisonId)) return offenders
+
+    val complexOffenders = complexityOfNeedAPI
```

Comment:
Maybe rename to highComplexityOffenders or even pass in the complexity threshold to this function?